service/worker/scanner/shardscanner/scanner_workflow.go

```diff
@@ -169,16 +165,7 @@ func (wf *ScannerWorkflow) Start(ctx workflow.Context) error {
 		}
 	}
 
-	activityCtx = getShortActivityContext(ctx)
-	summary := wf.Aggregator.GetStatusSummary()
-
-	return workflow.ExecuteActivity(activityCtx, ActivityScannerEmitMetrics, ScannerEmitMetricsActivityParams{
```

Comment:
This makes the scanner workflow non-deterministic. Even though we can manage this easily by restarting the workflow, it would be an annoyance for open-source users. We should either use GetVersion API or document this as a breaking change in the release notes to let users know that they should restart the scanner if it's running it. Given the small impact, I'd go with the release notes option.

Reply:
Good point Emrah, 
I will revert back this activity to not break the determinism here, but will reduce the list of metrics we emit instead